clients/ui/frontend/src/app/pages/modelCatalog/components/ModelCatalogActiveFilters.tsx

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import { ToolbarFilter, ToolbarLabelGroup, ToolbarLabel } from '@patternfly/react-core';
+import { isEnumMember } from 'mod-arch-core';
+import { ModelCatalogContext } from '~/app/context/modelCatalog/ModelCatalogContext';
+import {
+  ModelCatalogStringFilterKey,
+  MODEL_CATALOG_PROVIDER_NAME_MAPPING,
+  MODEL_CATALOG_LICENSE_NAME_MAPPING,
+  MODEL_CATALOG_TASK_NAME_MAPPING,
+  AllLanguageCodesMap,
+  MODEL_CATALOG_FILTER_CATEGORY_NAMES,
+  ModelCatalogProvider,
+  ModelCatalogLicense,
+  ModelCatalogTask,
+  AllLanguageCode,
+} from '~/concepts/modelCatalog/const';
+import { ModelCatalogFilterKey } from '~/app/modelCatalogTypes';
+
+type ModelCatalogActiveFiltersProps = {
+  filtersToShow: ModelCatalogFilterKey[];
+};
+
+const ModelCatalogActiveFilters: React.FC<ModelCatalogActiveFiltersProps> = ({ filtersToShow }) => {
+  const { filterData, setFilterData } = React.useContext(ModelCatalogContext);
+
+  const handleRemoveFilter = (categoryKey: string, labelKey: string) => {
+    if (isEnumMember(categoryKey, ModelCatalogStringFilterKey)) {
+      const currentValues = filterData[categoryKey];
+      if (Array.isArray(currentValues)) {
+        const newValues = currentValues.filter((v) => String(v) !== String(labelKey));
+        setFilterData(categoryKey, newValues);
+      }
+    }
+  };
+
+  const handleClearCategory = (categoryKey: string) => {
+    if (isEnumMember(categoryKey, ModelCatalogStringFilterKey)) {
+      setFilterData(categoryKey, []);
+    }
+  };
+
+  const getFilterLabel = (filterKey: ModelCatalogStringFilterKey, value: string): string => {
+    switch (filterKey) {
+      case ModelCatalogStringFilterKey.PROVIDER: {
+        return isEnumMember(value, ModelCatalogProvider)
+          ? MODEL_CATALOG_PROVIDER_NAME_MAPPING[value]
+          : value;
+      }
+      case ModelCatalogStringFilterKey.LICENSE: {
+        return isEnumMember(value, ModelCatalogLicense)
+          ? MODEL_CATALOG_LICENSE_NAME_MAPPING[value]
+          : value;
+      }
+      case ModelCatalogStringFilterKey.TASK: {
+        return isEnumMember(value, ModelCatalogTask)
+          ? MODEL_CATALOG_TASK_NAME_MAPPING[value]
+          : value;
+      }
+      case ModelCatalogStringFilterKey.LANGUAGE: {
+        return isEnumMember(value, AllLanguageCode) ? AllLanguageCodesMap[value] : value;
+      }
+      default:
+        return value;
+    }
+  };
+
+  return (
+    <>
+      {filtersToShow.map((filterKey) => {
+        // Only process string filter keys that are arrays
+        if (!isEnumMember(filterKey, ModelCatalogStringFilterKey)) {
+          return null;
+        }
+
+        const filterValues = filterData[filterKey];
+        if (!Array.isArray(filterValues) || filterValues.length === 0) {
+          return null;
+        }
```

Comment:
FYI this will need to be removed when we refactor this to also work for numeric filters, which will be required for the chip bar on the details page (and should probably wait for the ongoing refactors of the filter state)